haskell/haskell.bzl

```diff
@@ -138,6 +168,9 @@ def _mk_binary_rule(**kwargs):
         doc = "File containing `Main` module.",
       )
     ),
+    outputs = {
+      "repl": "%{name}-repl",
```

Comment:
Bazel conventions seem to pretty consistently favour [snake case](https://en.wikipedia.org/wiki/Snake_case). I think we should stick to Bazel conventions insofar as they exist for auto-generated target names.

Reply:
My logic here was:

* Names of (at least) library targets cannot contain underscores because they should be valid GHC package names (maybe now that we escape underscores it's not so important).
* So for consistency let's use dash here.

Snake case is understandable because `foo-bar` is not an identifier in Python/Skylark. But target names shouldn't necesserily be valid Skylark identifiers. In fact, at least in Numero I've seen a lot of targets named using dashes, such as `numero-doc`, etc.

Anyway, the choice of the separator seems not so important to me.